### ISSUE TITLE
[BUG] Error building tree using example data

### ISSUE THREAD

**Issue by Addicted-to-coding** [nextstrain/ncov#427]
Hi,
I'm trying to run a SARS-CoV-2 analysis using the example data with 418 sequences using the default command snakemake --profile profiles/default. 48% of the steps finish successfully, however, I get an error in building the tree as follows:
ERROR: b"ERROR: alignment.cpp:75: std::__cxx11::string& Alignment::getSeqName(int): Assertion `i >= 0 && i < (int)seq_names.size()' failed.\nERROR: STACK TRACE FOR DEBUGGING:\nERROR: 1   funcAbort()\nERROR: 2   ()\nERROR: 3   gsignal()\nERROR: 4   abort()\nERROR: 5   ()\nERROR: 6   ()\nERROR: 7   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 8   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 9   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 10   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 11   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 12   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 13   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 14   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 15   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 16   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 17   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 18   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 19   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 20   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 21   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 22   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 23   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 24   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 25   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 26   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 27   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 28   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 29   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 30   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 31   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 32   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 33   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 34   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 35   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 36   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 37   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 38   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 39   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 40   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 41   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 42   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 43   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 44   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 45   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 46   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 47   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 48   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 49   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 50   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 51   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 52   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 53   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 54   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 55   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 56   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 57   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 58   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 59   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 60   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 61   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 62   PhyloTree::assignLeafNames(Node*, Node*)\nERROR: 63   PhyloTree::readTreeString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)\nERROR: \nERROR: *** IQ-TREE CRASHES WITH SIGNAL ABORTED\nERROR: *** For bug report please send to developers:\nERROR: ***    Log file: results/global/subsampled_alignment-delim.fasta.log\nERROR: ***    Alignment files (if possible)\n/bin/bash: line 1:  5995 Aborted                 (core dumped) iqtree -ninit 2 -n 2 -me 0.05 -nt 2 -s results/global/subsampled_alignment-delim.fasta -m GTR -ninit 10 -n 4 > results/global/subsampled_alignment-delim.iqtree.log\n"
shell exited 134 when running: iqtree -ninit 2 -n 2 -me 0.05 -nt 2 -s results/global/subsampled_alignment-delim.fasta -m GTR -ninit 10 -n 4 > results/global/subsampled_alignment-delim.iqtree.log

Building a tree via:
	iqtree -ninit 2 -n 2 -me 0.05 -nt 2 -s results/global/subsampled_alignment-delim.fasta -m GTR -ninit 10 -n 4 > results/global/subsampled_alignment-delim.iqtree.log
	Nguyen et al: IQ-TREE: A fast and effective stochastic algorithm for estimating maximum likelihood phylogenies.
	Mol. Biol. Evol., 32:268-274. https://doi.org/10.1093/molbev/msu300

ERROR: TREE BUILDING FAILED
Please see the log file for more details: results/global/subsampled_alignment-delim.iqtree.log

Building original tree took 4.321278810501099 seconds
[Mon Jun  8 02:37:36 2020]
Error in rule tree:
    jobid: 15
    output: results/global/tree_raw.nwk
    log: logs/tree_global.txt (check log file(s) for error message)
    shell:
        
        augur tree             --alignment results/global/subsampled_alignment.fasta             --tree-builder-args '-ninit 10 -n 4'             --output results/global/tree_raw.nwk             --nthreads 2 2>&1 | tee logs/tree_global.txt
        
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: /home/vs/Downloads/COVID_Research/ncov/.snakemake/log/2020-06-08T023515.929219.snakemake.log

Could you help with this? I've not modified anything, I'm just using the files given and the default command.

**Comment by tsibley** [comment]
This looks like IQ-TREE crashed for some reason, which may not be something we can fix. The two log files `results/global/subsampled_alignment-delim.iqtree.log` and `logs/tree_global.txt` might contain more details. Can you upload those here for us?

**Comment by Addicted-to-coding** [comment]
Sure, pfa the files
[tree_global.txt](https://github.com/nextstrain/ncov/files/4750874/tree_global.txt)
[subsampled_alignment-delim.iqtree.log](https://github.com/nextstrain/ncov/files/4750878/subsampled_alignment-delim.iqtree.log)

**Comment by rneher** [comment]
I would try to reinstall IQ-tree. This problem is somewhere when IQ-tree reads your alignment and then crashes when an assertion fails.

**Comment by rneher** [comment]
I don't think this is a problem with augur or the ncov repo. could you try to run iqtree on your data outside of augur? Like
```
iqtree -ninit 2 -n 2 -me 0.05 -nt 2 -s results/global/subsampled_alignment-delim.fasta -m GTR -ninit 10 -n 4
```
and possible a simpler version:
```
iqtree -s results/global/subsampled_alignment-delim.fasta -m GTR
```

**Comment by trvrb** [comment]
I'm closing this issue now. @Addicted-to-coding please let us know if you're still running into this problem with the latest version of `ncov` and we can reopen if it's still a problem.

**Comment by pvanheus** [comment]
Hi there - I know this is a closed issued but I hit the same error. This [post](https://groups.google.com/g/iqtree/c/Gii__XoEIdU?pli=1) on the Iqtree forum discusses this issue. As Dr Bui states: "It means that you had a previous (unfinished) run and IQ-TREE loaded the checkpoint file. From what you wrote below, I’m guessing that you now change the alignment file (-s option). This causes incompability between the checkpoint and the alignment."

This *does*, I think, describe the circumstances in which I encountered this: I had started a Nextstrain build, then terminated it for some reason, added some more sequences to the input data and re-started the build. I can imagine some "checkpoint" file being left behind in the `results` directory that contracted the alignment file. I subsequently deleted the results directory and re-started the build. It has now proceeded past the tree-building step with no errors. 

I haven't been able to prove that this is the source of the error but I'm just leaving this theory here for further investigation...
